### PR TITLE
fix: fix failpoint tests and add to CI

### PR DESCRIPTION
**Pull request by hzxa21**
This PR fixes the failure injection tests introduced in #1564 and modifies the CI workflow to run failiure injection tests.

## Refer to a related PR or issue link (optional)
#1221

### CODE REVIEW

**Comment by hzxa21** [comment]
This PR depends on the bug fix in #2381

File: src/storage/src/storage_failpoints/test_iterator.rs [hunk 1]
```diff
@@ -154,12 +159,16 @@ async fn test_failpoint_merge_invalid_key() {
...
-    let tables = vec![Arc::new(table0), Arc::new(table1)];
+    let tables = vec![table0, table1];
+    let cache = create_small_table_cache();
     let mut mi = MergeIterator::new(
         tables
             .iter()
             .map(|table| -> Box<dyn HummockIterator<Direction = Forward>> {
-                Box::new(SSTableIterator::new(table.clone(), sstable_store.clone()))
+                Box::new(SSTableIterator::new(
+                    cache.insert(table.id, table.id, 1, Box::new(table.clone())),
```
Comment (wenym1) [inline src/storage/src/storage_failpoints/test_iterator.rs:169]: It's better to use `sstable_store.sstable(table.id)?`

File: src/storage/src/storage_failpoints/test_iterator.rs [hunk 2]
```diff
@@ -198,13 +207,14 @@ async fn test_failpoint_backward_merge_invalid_key() {
...
-    let tables = vec![Arc::new(table0), Arc::new(table1)];
+    let tables = vec![table0, table1];
+    let cache = create_small_table_cache();
     let mut mi = BackwardMergeIterator::new(
         tables
             .iter()
             .map(|table| -> Box<dyn HummockIterator<Direction = Backward>> {
                 Box::new(BackwardSSTableIterator::new(
-                    table.clone(),
+                    cache.insert(table.id, table.id, 1, Box::new(table.clone())),
```
Comment (wenym1) [inline src/storage/src/storage_failpoints/test_iterator.rs:217]: Same as above

File: src/storage/src/storage_failpoints/test_iterator.rs (hunk 1 above)
Comment (wenym1) [reply src/storage/src/storage_failpoints/test_iterator.rs:169]: Same for all following.

File: src/storage/src/storage_failpoints/test_hummock.rs [hunk 3]
```diff
@@ -79,10 +77,11 @@ async fn test_failpoint_state_store_read_upload() {
     local_version_manager
         .refresh_version(meta_client.as_ref())
         .await;
-
+    // clear block cache
+    sstable_store.clear_block_cache();
     fail::cfg(mem_read_err, "return").unwrap();
 
-    let result = hummock_storage.get(b"bb".as_ref(), 2).await;
+    let result = hummock_storage.get(&anchor, 2).await;
```
Comment (wenym1) [inline src/storage/src/storage_failpoints/test_hummock.rs:84]: Seems that anchor is `"aa"`, while the original one is `"bb"`. Is this consistent with the original code?

File: .github/workflow-template/jobs/compute-node-test.yml [hunk 4]
```diff
@@ -47,6 +47,10 @@ jobs:
         run: |
           cargo doc --document-private-items --no-deps
 
+      - name: Run rust failpoints test
```
Comment (skyzh) [inline .github/workflow-template/jobs/compute-node-test.yml:50]: Shall we enable clippy check for failpoints? (cargo clippy should be run with --all-features)

**Review by skyzh — APPROVED** [review]
Rest LGTM

File: .github/workflow-template/jobs/compute-node-test.yml (hunk 4 above)
Comment (hzxa21) [reply .github/workflow-template/jobs/compute-node-test.yml:50]: good point. added.

File: src/storage/src/storage_failpoints/test_hummock.rs (hunk 3 above)
Comment (hzxa21) [reply src/storage/src/storage_failpoints/test_hummock.rs:84]: I think it is consistent. This test is testing get with object store returuing an error on read. `"bb"` is an non-existent key and it is possible that bloom filter will filter it out before reaching object store so it is better to use an existent key here. cc @xxhZs

**Review by wenym1 — APPROVED** [review]
LGTM

File: src/storage/src/storage_failpoints/test_hummock.rs (hunk 3 above)
Comment (xxhZs) [reply src/storage/src/storage_failpoints/test_hummock.rs:84]: yes , I think it is consistent too

Comment (wenym1) [reply src/storage/src/storage_failpoints/test_hummock.rs:84]: I suggest we explicitly disable bloom filter when failpoints is enabled, also for cache policy. `failpoints` provides a function `fail::has_failpoints` to check whether it is enabled (simply a `cfg!(feature = "failpoints"))`.

**Pull request merged**